### PR TITLE
Handle image data paste early

### LINKED ISSUE #3537: Allow pasting of images from Clipboard
One of my favorite features in GitHub is being able to paste images from your clipboard directly into the editor. This would be a great future enhancement for Gutenberg.

### PR DESCRIPTION
**Pull request by ellatrix**
## Description
Fixes #3537. In Chrome, image data paste fails because the `onPastePreProcess` event does not happen. TinyMCE has different paste logic and image data handling, so it would be better if we handle data paste early instead of letting it pass unnecessarily through TinyMCE.

## How Has This Been Tested?
Copy an image in Chrome and paste it in Gutenberg.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by mcsf** [comment]
Preliminary testing reveals this `TypeError` in `tinymce`:

gutenberg-paste-image

screen shot 2018-01-05 at 14 23 59

**Comment by iseulde** [comment]
@mcsf I cannot reproduce this. Is this Chrome?

File: blocks/editable/index.js [hunk 1]
```diff
@@ -288,12 +288,27 @@ export default class Editable extends Component {
...
 			// Allows us to ask for this information when we get a report.
-			window.console.log( 'Received item:\n\n', item );
+			window.console.log( 'Received item:\n\n', blob );
 
-			const blob = item.getAsFile ? item.getAsFile() : item;
+			if ( isEmpty && this.props.onReplace ) {
+				this.props.onReplace( content );
+			} else {
+				// Necessary to get the right range.
+				// Alse done in the TinyMCE paste plugin.
```
Comment (mtias) [inline blocks/editable/index.js]: little typo "alse".

Comment (iseulde) [reply blocks/editable/index.js]: Fixed, thanks!

**Comment by mcsf** [comment]
@iseulde: yes, Chrome. See:

gutenberg-chrome-paste-image

**Comment by iseulde** [comment]
@mcsf Cool, seeing the error now. :)

**Comment by iseulde** [comment]
It's finny that this error happens at code where there is the following comment:

```js
// WebKit/Blink might clone the div so
// lets make sure we remove all clones
// TODO: Man o man is this ugly. WebKit is the new IE! Remove this if they ever fix it!
```

**Comment by iseulde** [comment]
We're doing `event.preventDefault()` and then down the line this is causing the error:

```js
if (e.isDefaultPrevented() || isBrokenAndroidClipboardEvent(e)) {
    pasteBin.remove();
    return;
}
```

So at first sight it seems like a bug in TinyMCE.

**Comment by iseulde** [comment]
See https://wordpress.slack.com/archives/C0UCMQP0F/p1515587384000026.

I added a timeout (like the one for splitting content) to allow the pastern to be removed before we replace the block (and the editor).

**Comment by iseulde** [comment]
@mcsf How is it looking now?

**Comment by mcsf** [comment]
> @mcsf How is it looking now?

Yay, it works now, thanks!

**Review by mcsf — APPROVED** [review]

**Comment by iseulde** [comment]
Thanks!

**Pull request merged**